src/Spanner/Session/CacheSessionPool.php

```diff
@@ -777,4 +764,63 @@ private function validateConfig()
             );
         }
     }
+
+    /**
+     * Delete the provided sessions.
+     *
+     * @param array $sessions
+     */
+    private function deleteSessions(array $sessions)
+    {
+        $calls = [];
+
+        foreach ($sessions as $session) {
+            $calls[] = $this->database->connection()
+                ->deleteSessionAsync([
+                    'name' => $session['name'],
+                    'database' => $this->database->name()
+                ]);
+        }
+
+        foreach ($calls as $call) {
+            $call->wait();
```

Comment:
Good question. Yes, the deletes wouldn't finish until the process ended. WDYT about making the usage of `register_shutdown_function` optional, and turned on by default?

Hopefully I'll find it's possible we don't need to call `wait` and we can avoid this altogether though :).